### DISCUSSION TITLE
Unable to control volume

### DISCUSSION THREAD

**Discussion by lwpersonal** [MonitorControl/MonitorControl#1597] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Question

On my computer, I could not control the volume of the monitor with MonitorControl, but I could with the brightness control.

I turned on mac-related accessibility and restarted the computer, but it still doesn't work.

The following is the adjustment interface that comes with the software. If I were to use the adjustment interface that comes with my Mac computer, none of the interface would be displayed. Although there is interface feedback but the volume has not changed.

I don't know what to do to make the volume adjustment take effect, I searched for a long time but couldn't find a solution, I hope I can help to look at this problem. thx.

 <img width="395" alt="image" src="https://user-images.githubusercontent.com/26055361/235831330-4bd1c408-7f77-4f34-8f93-ea43b10ef451.png">

<img width="541" alt="image" src="https://user-images.githubusercontent.com/26055361/235831505-920b844f-d5a4-46a7-9096-037676e94d58.png">

<img width="730" alt="image" src="https://user-images.githubusercontent.com/26055361/235831532-fd57821b-a0bc-43ef-944d-6026045658f8.png">

**Comment by lwpersonal** [comment]
This is the information configuration of my computer and I am using the latest MonitorControl software version 4.1.0.

<img width="280" alt="image" src="https://user-images.githubusercontent.com/26055361/235831762-beb7513a-4f3c-400f-a1c8-c4657841151f.png">

**Comment by segalek** [comment]
Hello,
I have similar problem - couldn't control the volume of the monitor with MonitorControl. I have monitor Philips 346P1C
With the brightness control everything is OK

**Comment by lwpersonal** [comment]
> I'm using INNOCN.

I am using INNOCN and now I can only adjust the volume with the mouse and it is very broken.

**Comment by davorbettercare** [comment]
I got Philips 346P1C connected via USB-C to macbook 14 pro M2.
But in my case, I don't even get option for volume control, only brightness (which works).

**Comment by balzercomp** [comment]
I've a similar problem. I'm using an LG 49WL95C  connected via USB C. Since I installed Monitor Control instead of the Light version, the volume controls are displayed on the sceen, but changing the volume doesn't have any effect on the real volume.

**Comment by davordragic** [comment]
> I got Philips 346P1C connected via USB-C to macbook 14 pro M2. But in my case, I don't even get option for volume control, only brightness (which works).

Update. Lite version does not give an option, while this from github, I get volume bar, but same as others, no actual volume change.

**Comment by waydabber** [comment]
HI all,

some displays just don't support volume control over DDC - for these displays the volume slider will be ineffective.

The App Store "Lite" version does not support DDC (as Apple does not allow submissions with that functionality) so that's why there is no volume controls in that version of the app.

DDC on M2 Pro systems are not supported by MonitorControl currently. The brightness slider on these Macs will fall back to software dimming in MonitorControl. You can use [BetterDisplay](https://github.com/waydabber/BetterDisplay) for full DDC support on these systems.

**Comment by Alraemon** [comment]
> HI all,
> 
> some displays just don't support volume control over DDC - for these displays the volume slider will be ineffective.
> 
> The App Store "Lite" version does not support DDC (as Apple does not allow submissions with that functionality) so that's why there is no volume controls in that version of the app.
> 
> DDC on M2 Pro systems are not supported by MonitorControl currently. The brightness slider on these Macs will fall back to software dimming in MonitorControl. You can use [BetterDisplay](https://github.com/waydabber/BetterDisplay) for full DDC support on these systems.

I am using macOS 13.4, MacBook Pro with M1 Pro. Same issue here. Brightness control is ok, but volume control is not effective, though the slider shows. The setting panel shows that it's using DDC to control the monitor. And I have confirmed BetterDisplay can actually control the volume with DDC. So it might be the software issue?

**Comment by boabdilperez** [comment]
I had the same issue, turned out to be incompatibility with the M2 Pro Macbook. Switching to [BetterDisplay](https://github.com/waydabber/BetterDisplay) worked for me. 

MonitorControl is a much simpler and cleaner interface, if/when they update for M2 support I'll probably switch back.

**Comment by azizoid** [comment]
I have Samsung M7 monitor + Mac Mini m2.
I can control the volume. Whenever i change the volume via keyboard. slider moves, but volume does not change

**Comment by adib** [comment]
I'm reporting the same issue of being unable to control the monitor's volume control. The mute button on the keyboard is also non-functional. This is even though pressing the volume control and mute buttons on the keyboard shows the corresponding feedback HUD on the screen.

Nevertheless, the monitor's hardware joystick is able to change the audio volume of speakers connected through the monitor.

Configuration:

 - Monitor: [LG 27UL500-W](https://www.lg.com/us/monitors/lg-27UL500-W) connected via the built-in HDMI.
 - M1 (basic) Mac Mini, 2020.
 - macOS 13.4

**Comment by waydabber** [comment]
Guys, again: the HDMI port of any Mini (Intel or M1/M2) since the 2018 model year is not supported for DDC with MonitorControl (hardware backlight, contrast and volume control won't work). Use apps like [BetterDisplay](https://betterdisplay.pro/ - all DDC features are free and uncrippled) for control over the built-in HDMI port of any Mac until this feature is added to MonitorControl by the community.

**Comment by mikemklee** [comment]
Also encountering same issue as described by @adib, on my Macbook Pro (M1 Pro, 2021, 16-inch).

 HUD appears when volume keys are pressed on keyboard, but no actual change from HDMI audio output from monitor.

Manually adjusting the volume with physical controls on the monitor does work.

Interestingly, this seems to work on my other Macbook Pro (M2, 2022, 13-inch).

**Comment by waydabber** [comment]
@mikemklee, HDMI won't work on any M1 MBPs. Use BetterDisplay until the support is added in MC.

**Comment by mikemklee** [comment]
@waydabber I did try using `BetterDisplay` instead, but observed the same behavior where the volume didn't actually change despite the HUD showing. Unless I had to configure the app additionally to get it working.

Out of curiosity, is it a hardware issue that DDC won't work on M1 MBPs?
Or is it something that needs to be updated in `MonitorControl`?

Wondering what exactly is missing, and whether it's something I can contribute towards adding.

**Comment by waydabber** [comment]
Hi @mikemklee - DDC does work on M1 MBPs. MC supports it via USB-C only but BD has support for the built-in HDMI port as well. But many displays do not support DDC volume control so it might be that - did you get it working with other devices?

**Comment by mikemklee** [comment]
@waydabber ahh interesting.

thanks for the heads up on MC only supporting DDC via USB-C.

I was previously connecting my M1 MBP to the monitor via the HDMI port.
I changed the connection to USB-C using a HDMI <> USB-C dongle, and now DDC seems to work (with BD)! 🙌

**Comment by waydabber** [comment]
Nice! Note: using this dongle it should work with MC as well. BD should support DDC over the built-in HDMI port as well so if it doesn't work, I am not sure what could be the problem with that. But the main thing is, you made it work! :)

**Comment by azizoid** [comment]
switched to usb-c to usb-c cable, usb3.2.
same problem, slide moves, but volume does not change. i can change the volume only using the repote controller of the monitor
Mac Mini M2
Samsung M7

**Comment by Cyfine** [comment]
My INNOCN 27M2U monitor's volume can be adjusted directly through the built-in volume slider in Windows 11. However, when using MonitorControl in macOS (MBA M2, 2022), the volume cannot be controlled. It is possible that certain monitor models may employ Windows-specific protocols for the convenience of Windows users, which are not supported by MonitorControl.

**Comment by waydabber** [comment]
No. Windows does not change the display's hardware audio levels (using DDC) but changes the output level of digital audio stream when a display is connected via HDMI or DisplayPort. There are solutions for macOS that do the same (like eqMac) using virtual audio devices - it's just not a built-in feature of the OS. Some displays are simply not compatible with DDC audio volume control - for these displays DDC volume control won't work even if a supported display connection is used on a supported Mac.

**Comment by evertonspader-tomtom** [comment]
I have two Macbooks 16", one with M1 Pro and the other with M2 Pro. I am connecting them with a monitor Dell U3421WE via USB-C with PD. The Macbook with M1 Pro has all controls enabled for brightness and sound. The one with M2 Pro doesn't have the volume control enabled. I am running both with MonitorControl Version 4.1.0 Build 7034 and what I notice is that the option for "Use hardware DDC control" is not possible to be enabled on the M2 Pro as it is on the M1 Pro.
Since I saw that more people are having the same issue with different monitors and processors, I can only think of a software issue.

* Controls on M1 Pro
![Screenshot 2023-06-23 at 12 13 45](https://github.com/MonitorControl/MonitorControl/assets/136707241/8cf2e197-b9d8-4204-bfda-346400135612)
![Screenshot 2023-06-23 at 12 20 36](https://github.com/MonitorControl/MonitorControl/assets/136707241/e0fbef38-7e39-4dc9-a748-986db9a29b26)

* Controls on M2 Pro
![image](https://github.com/MonitorControl/MonitorControl/assets/136707241/5e11c1ac-f653-4fc3-a984-dde8ce660a32)
![image](https://github.com/MonitorControl/MonitorControl/assets/136707241/87ab6911-048f-4bba-964d-bf64ccde9a59)

**Comment by dagassa** [comment]
<img width="1013" alt="Bildschirmfoto 2023-07-04 um 15 17 08" src="https://github.com/MonitorControl/MonitorControl/assets/7697234/7520e3b4-e2c7-434c-a7c2-21a2a2afd6b7">

Unfortunately the same problem with my Mac mini over usb c. My Intel Macbook Pro  can control the volume of the LG display over HDMI without any issues.

**Comment by bverkron** [comment]
I have this issue as well. This DID work when I was running Big Sur but now that I've upgraded to Ventura it no longer works. My monitor is a Dell S2721DGF.  I'm running an intel based MacBook Pro 15" 2014

**Comment by rh-01** [comment]
Same issue here, with a Mac Mini M2 and a Dell U2520D (monitor connected via USB-C, external loudspeaker connected via headphone jack on monitor). The volume popup display moves when I press the volume keys, as expected, but volume does not change (see below).

Note: I am able to control the volume of the external speaker with the volume control of the Apple Music App, so it can't be a hardware issue...

here the display changes when I press the volume keys:

![image](https://github.com/MonitorControl/MonitorControl/assets/96235063/2738750b-7551-4a9b-8b22-1112df0ee3fe)

here, too:

![image](https://github.com/MonitorControl/MonitorControl/assets/96235063/4ea2b9af-6de8-4a22-b440-9fd9552a7854)

but here (MacOS menu bar volume control) it is blocked:

![image](https://github.com/MonitorControl/MonitorControl/assets/96235063/05c0fb5a-c6fa-499e-89ce-834b6d280fe3)

**Comment by Addlai** [comment]
Same problem on Mac Mini M1 2020 using LG 43in 4K Ultrafine over HDMI.

**Comment by waydabber** [comment]
MonitorControl does not support DDC over the built-in HDMI on the 2018 mini and M1 (and entry level M2) Macs. [BetterDisplay](https://betterdisplay.pro) implements support for all.

**Comment by dagassa** [comment]
@waydabber thank you, probably time to switch...

**Comment by heliumllc** [comment]
It's been a while and still no fix whatsoever. Looks like this app is abandoned by devs..

**Comment by azizoid** [comment]
will add comment here, so they will not think to archive this issue for inactivity

**Comment by limeandcoconut** [comment]
For anyone else ending up here trying to figure out why MonitorControl or [BetterDisplay](https://github.com/waydabber/BetterDisplay) doesn't work. Your monitor's internal settings may have DCC turned off. My EW3280u has it under System>DCC/CI.

**Comment by 011V32** [comment]
Hello @waydabber

What I don't understand: Why isn't the HDMI support added if the project isn't dead?
If we should rather use BetterDisplay, why not show it at the top of the repo and put this repo into maintenance mode?

I only stumbled upon this topic after I installed the application. (Of course I could have read it in the info beforehand...) But I assumed that errors or such functions would at least appear in a maintained project and I would have no problem waiting for them.
thx

**Comment by waydabber** [comment]
Hi @011V32,

The question is legit. This issue is unrelated to the project's status. For M1, entry level M2 and the Intel 2018 mini HDMI support was not open sourced (yet), that's why it is not added to MonitorControl. The reason is simple - there are other paid (not even freemium) apps that would instantly snatch this so it just does not make sense to publish the method (however it is not rocket science, an interested developer with some digging can - I think - figure out how to do it). I made this feature available completely free in [BetterDisplay](https://betterdisplay.pro), no Pro license required. I also added M2/M3 support to MonitorControl quite recently and open-sourced that change. 😄 

Note: HDMI is supported for M2 Pro/Max and M3 in MonitorControl.

**Comment by 011V32** [comment]
Hi @waydabber,

Thank you for your feedback. I appreciate what you do in your free time. 👍 
What I don't understand about not publishing: It is also included in BetterDisplay, which means it is already public, or am I not understanding something here?
Will support still be coming in future, or can we no longer count on it in this app?
Thank you for your commitment.

**Comment by waydabber** [comment]
Hi @011V32 - BetterDisplay is free for these features but not open-source, so the implementation itself is not public. MonitorControl is free and open source as well. Regarding MonitorControl, the team decided that the app is feature complete so I don't think there will be any major new releases (there might still be service releases though).

**Comment by TheNew1997** [comment]
Hi, for me the volume control and mute working, but when I used unmute after mute with key control, the sound not working. On monitor the volume presented as unmute. I must physical on monitor mute and unmute volume and after that it is working. MacBook Pro M1 2020, macOS Sonoma 14.3.1 and monitor 34" Samsung ViewFinity S65UC.

**Comment by waydabber** [comment]
Hi @TheNew1997 

With BetterDisplay you have more control on how mute/unmute works. Try to experiment with that.

<img width="994" alt="Screenshot 2024-02-29 at 08 45 13" src="https://github.com/MonitorControl/MonitorControl/assets/37590873/5d87d446-f207-4810-8da0-03ffcbf01803">

More info: https://github.com/waydabber/BetterDisplay/issues/1248

**Comment by mb10977** [comment]
> BetterDisplay

Same here, using the ViewFinity S9 volume controls are not working.

**Comment by aligkts** [comment]
Same here, using Samsung M7

**Comment by aspargusgit** [comment]
> > HI all,
> > some displays just don't support volume control over DDC - for these displays the volume slider will be ineffective.
> > The App Store "Lite" version does not support DDC (as Apple does not allow submissions with that functionality) so that's why there is no volume controls in that version of the app.
> > DDC on M2 Pro systems are not supported by MonitorControl currently. The brightness slider on these Macs will fall back to software dimming in MonitorControl. You can use [BetterDisplay](https://github.com/waydabber/BetterDisplay) for full DDC support on these systems.
> 
> I am using macOS 13.4, MacBook Pro with M1 Pro. Same issue here. Brightness control is ok, but volume control is not effective, though the slider shows. The setting panel shows that it's using DDC to control the monitor. And I have confirmed BetterDisplay can actually control the volume with DDC. So it might be the software issue?

The app BetterDisplay solved the issue of the both, bright and volume control. Thank you!

**Comment by greatgifts** [comment]
Same here, using Samsung M7

**Comment by warpdesign** [comment]
Same here: Lenovo L27q-10 connected on M1 Pro MacBook Pro via HDMI.

Slider moves, but sound volume does not change. Same thing with BetterMonitor.
